### PR TITLE
Align Face To Build-Plate

### PR DESCRIPTION
**Pull request by rburema**
Alt-click to select a face. See the (identically named) 'feature_bottom_face' branch in Uranium for more indepth information.

You'll need https://github.com/Ultimaker/Uranium/pull/519 to get this to work!

_ edit: CURA-6745 _

### CODE REVIEW

File: cura/CuraActions.py [hunk 1]
```diff
@@ -73,6 +75,36 @@ def centerSelection(self) -> None:
...
+    @pyqtSlot()
+    def bottomFaceSelection(self) -> None:
+        selected_face = Selection.getSelectedFace()
+        if not selected_face:
+            Logger.log("e", "Bottom face operation shouldn't have been called without a selected face.")
+            return
+
+        original_node, face_id = selected_face
+        meshdata = original_node.getMeshDataTransformed()
+        if not meshdata or face_id < 0 or face_id > 0x10001:
```
Comment (LipuFei) [inline cura/CuraActions.py]: I'm not familiar with this part, so perhaps document what `face_id < 0` and `face_id >0x10001` mean? Those are magic numbers to me.

File: plugins/SolidView/SolidView.py [hunk 2]
```diff
@@ -139,6 +139,10 @@ def beginRendering(self):
                             shade_factor * int(material_color[5:7], 16) / 255,
                             1.0
                         ]
+
+                        # Color the currently selected face-id, 0x10001 is certain to be greater than the largest ID.
+                        face = Selection.getSelectedFace()
+                        uniforms["selected_face"] = 0x10001 if not face or node != face[0] else face[1]
```
Comment (LipuFei) [inline plugins/SolidView/SolidView.py]: OK, so it's documented here. Perhaps make the magic number `0x1001` a constant with a descriptive name?

File: cura/CuraActions.py (hunk 1 above)
Comment (rburema) [reply cura/CuraActions.py]: Fixed!

File: plugins/SolidView/SolidView.py (hunk 2 above)
Comment (rburema) [reply plugins/SolidView/SolidView.py]: Fixed!

**Comment by smartavionics** [comment]
BTW, you do know that Linux systems use Alt-drag to move windows on the desktop and so Alt-click probably won't be delivered to the app?

**Comment by LipuFei** [comment]
> 
> BTW, you do know that Linux systems use Alt-drag to move windows on the desktop and so Alt-click probably won't be delivered to the app?

@nallath @Ghostkeeper Could you test this?

**Comment by smartavionics** [comment]
Maybe on Linux you will need to use the right hand Alt key instead?

**Comment by rburema** [comment]
Hi @smartavionics Thanks for the heads-up! We already decided that the alt-click is a bit awkward anyway, so now the main way to do this is selecting a checkbox in the RotationTool instead.

**Pull request merged**